drawers/cBuildingListDrawer.cpp

```diff
@@ -1,19 +1,18 @@
-#include "../include/d2tmh.h"
 #include "cBuildingListDrawer.h"
 
+#include "cAllegroDrawer.h"
+#include "d2tmc.h"
+#include "data/gfxinter.h"
+#include "player/cPlayer.h"
 
 #include <allegro.h>
 
 cBuildingListDrawer::cBuildingListDrawer(cPlayer *player) :
-    m_textDrawer(new cTextDrawer(game_font)),
+    m_textDrawer(game_font),
```

Comment:
Uhm, not really. This constructs `m_textDrawer` in-place inside `cBuildingListDrawer` so-to-speak. It's object memory is layout inside the object memory of `cBuildingListDrawer`. What you typed uses the copy constructor (unless that is optimized out - never know what the optimizer can do nowadays). So that would construct `cTextDrawer` on the stack and then copy (or move if it can) it into `cBuildingListDrawer`.

TL;DR: this is a way to use data member with non-default constructors.